woocommerce/woocommerce-gutenberg-products-block
# Product Collection: Add 'on sale' filter and enhance settings management in product collection block

This PR introduces several changes to the product collection block. 
- First, it adds a new 'Show only products on sale' filter that can be used to display only the products that are currently on sale. 
<img width="281" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/f91284c2-347f-41a8-a164-d70e75d7c96c">

- It also refactors the settings management in the product collection block to use the experimental ToolsPanel component from WordPress, which provides a more flexible and intuitive way to manage block settings. 

This should enhance the flexibility and user-friendliness of the product collection block.

Fixes #9361 

### Testing

1. Create a new post
2. Add Product Collection block
3. Verify that the sidebar displays a 'On sale' setting when the Product Collection block is selected.
<img width="281" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/f91284c2-347f-41a8-a164-d70e75d7c96c">

4. Toggle "Show only products on sale"
   - Confirm that the change is immediately reflected in the editor & showing only products on sale.
5. Save and publish the post.
   - Verify that the same change is reflected on the frontend.
6. Also, verify that in Editor, filters can be reset using Reset/Reset all button:
<img width="284" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/b76fdb4f-fa6e-4c59-8a49-3c605279ad2e">



### WooCommerce Visibility




### Changelog

> Product Collection: Add 'on sale' filter and enhance settings management in product collection block
